### PR TITLE
Follow up #2617

### PR DESCRIPTION
**Pull request by ericpre**
Fix some breaking changes introduced in #2617 for the private API `_map_iterate`.

### CODE REVIEW

File: hyperspy/_signals/lazy.py [hunk 1]
```diff
@@ -588,7 +591,14 @@ def _map_iterate(self,
         nav_chunks = old_sig._get_navigation_chunk_size()
         args = ()
         arg_keys = ()
+
         for key in iterating_kwargs:
+            if not isinstance(iterating_kwargs[key], BaseSignal):
+                iterating_kwargs[key] = BaseSignal(iterating_kwargs[key].T).T
+                warnings.warn(
```
Comment (francisco-dlp) [inline hyperspy/_signals/lazy.py:598]: passing array**s** as keyword argument**s** ?

Use Signal instead -> pass signal instances instead?

File: hyperspy/signal.py [hunk 2]
```diff
@@ -4607,7 +4607,8 @@ def _map_iterate(
             A tuple with structure (('key1', value1), ('key2', value2), ..)
             where the key-value pairs will be passed as kwargs for the
             function to be mapped, and the values will be iterated together
-            with the signal navigation.
+            with the signal navigation. The value needs to be signal because
```
Comment (francisco-dlp) [inline hyperspy/signal.py]: signal -> a signal instance?

**Review by francisco-dlp — COMMENTED** [review]
Except for the couple of suggestions above, LGTM!

**Comment by ericpre** [comment]
Thanks, done.

**Pull request merged**